azurerm/internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -258,6 +258,12 @@ func resourceArmApiManagementApiCreateUpdate(d *schema.ResourceData, meta interf
 			},
 		}
 		wsdlSelectorVs := importV["wsdl_selector"].([]interface{})
+
```

Comment:
I saw you linked this to #6795, but I don't think that it's the cause for what I reported. It also affected `swagger_json` formats.

I _think_ it might have something to do with the whole if-statement that this code is in. The comment on line 241-242 is:
> 	// If import is used, we need to send properties to Azure API in two operations.
>	// First we execute import and then updated the other props.

This was added when the underlying API was 2018-XX-XX. and I'm guessing that it was a workaround for a bug back then. I could not find anything in the REST API doc. that stated it has to be two requests. 

Could it be an idea to do this in one request instead of two? Should not be very difficult to merge the request into one.

Reply:
Hi @tnicolaysen ,for issue https://github.com/terraform-providers/terraform-provider-azurerm/issues/6795, I have tried to add `wsdl_selector` in the code, and it's created with success. Do I miss anything?

Reply:
I can try to make a build from this branch and see if it solves my problem. But, it is still weird that the code performs the update in two steps.

Reply:
I have tested this branch with my setup.

* Creation of new OpenAPI APIs work now.
* Update of existing APIs work.
* Switch from swagger_json to openapi on an existing API worked.
* The primary/secondary key issue is confirmed fixed for me. (Not relevant to this PR).

I also got this expected error:
```
Error: `wsdl_selector` is required when content format is `wsdl` in API Management API "xxx-api"

  on main.tf line 163, in resource "azurerm_api_management_api" "xxx":
 163: resource "azurerm_api_management_api" "xxx" {
```

I didn't know that it was required. It's not required when importing a WSDL through Azure Portal. Maybe they just default to the first or something.

Reply:
hi @tnicolaysen , yes, I have split the two problems in two PRs. Would you like to try if the below code works for you? And on portal, `wsdl_selector` is required as the screenshot below.
```
resource "azurerm_api_management_api" "example" {
  api_management_name = data.azurerm_api_management.apim.name
  resource_group_name = data.azurerm_api_management.apim.resource_group_name

  display_name      = "Example API"
  name              = "example-api"
  path              = "example-api"
  protocols         = ["https"]
  revision          = "1"
  service_url       = "http://example.org"
  soap_pass_through = true

  // NOTICE: It partially works when not having an import, but it's just one of the issues
  import {
    content_format = "wsdl"
    content_value = file("example.wsdl")

 wsdl_selector {
      service_name  = "Calculator"
      endpoint_name = "CalculatorHttpsSoap11Endpoint"
    }
  }
}
```

![image](https://user-images.githubusercontent.com/56525716/82967319-d60e6a80-9ffe-11ea-8ca1-0172377a4018.png)

Reply:
Looks like we're seeing different UIs. Might have something to do with the underlying API-M version and when it was provisioned. 

![image](https://user-images.githubusercontent.com/194898/82994218-17ece000-a002-11ea-8c31-27998187ff40.png)

I tested with code that was similar to what you showed, and it worked for me.